frontend/src/components/dashboard/settings/api-key/CreateUpdateModal.tsx

```diff
@@ -0,0 +1,168 @@
+      ...contactList.slice(0, index),
+      value.toLowerCase(),
+      ...contactList.slice(index + 1),
+    ])
+  }
+  function addNewContact() {
+    setContactList([...contactList, ''])
+  }
+  function removeContact(index: number) {
+    setContactList([
+      ...contactList.slice(0, index),
+      ...contactList.slice(index + 1),
+    ])
+  }
+
+  function presubmissionCheck(): string | undefined {
+    if (isLoading) {
+      return 'Another submission is currently being requested'
+    }
+    if (contactList.length < 1) {
+      return 'You must enter at least 1 contact email address'
+    }
+    const emailRegex =
+      /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i
+    if (contactList.some((v) => !emailRegex.test(v))) {
+      return 'One of the contact email addresses is invalid'
+    }
+  }
+
+  async function handleCreateSubmit(e: FormEvent) {
+    e.preventDefault()
+    const presubmissionErr = presubmissionCheck()
+    if (presubmissionErr) {
+      displayError(presubmissionErr)
+      return
+    }
+    try {
+      setIsLoading(true)
+      const apiKey = await generateApiKey({
+        label: keyLabel,
+        notificationAddresses: contactList,
+      })
+      await onSuccess(apiKey)
+    } catch (e) {
+      displayError('Something went wrong. Please try again later!')
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  async function handleUpdateSubmit(e: FormEvent) {
+    e.preventDefault()
+    const presubmissionErr = presubmissionCheck()
+    if (presubmissionErr) {
+      displayError(presubmissionErr)
+      return
+    }
+    if (!isUpdateMode) {
+      return
+    }
+
+    try {
+      setIsLoading(true)
+      const apiKey = await updateApiKey({
+        id: originalApiKey.id,
+        notificationAddresses: contactList,
+      })
+      await onSuccess(apiKey)
+    } catch (e) {
+      displayError('Something went wrong. Please try again later!')
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const submitButtonText = {
+    loading: isUpdateMode ? 'Updating Key' : 'Generating Key',
+    default: isUpdateMode ? 'Update Key' : 'Updating Key',
+  }
+  return (
+    <form onSubmit={isUpdateMode ? handleUpdateSubmit : handleCreateSubmit}>
+      <div>
+        <h2>
+          <label htmlFor="apiKeyLabel">Create a new API key</label>
+        </h2>
+        <h4>Key Label</h4>
+        <p>
+          The label you choose for your API key is be permanent and cannot be
+          changed once it has been created.
+        </p>
+        <TextInput
+          id="apiKeyLabel"
+          type="text"
+          placeholder="The label with which you can identify your API key with"
+          value={keyLabel}
+          onChange={setKeyLabel}
+          disabled={isUpdateMode}
+        ></TextInput>
+        <h4>Contact Emails</h4>
+        <p>
+          Please provide the email contacts in the event we need to reach out to
```

Comment:
**[copy]** Please enter email addresses that will receive updates regarding this API key (e.g. reminders to rotate expiring key, information about new features etc.). It is recommended to keep these email addresses updated so that critical updates will not be missed.